igwn_auth_utils/requests.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Cardiff University
+# Distributed under the terms of the BSD-3-Clause license
+
+"""Python Requests interface with IGWN authentication
+
+This is heavily inspired by Leo Singer's excellent `requests-gracedb` package.
+"""
+
+__author__ = "Duncan Macleod <duncan.macleod@ligo.org>"
+__credits__ = "Leo Singer <leo.singer@ligo.org>"
+
+from os import getenv
+from netrc import netrc
```

Comment:
I suggest using my `safe-netrc` package instead, which I developed in response to an LSC computing security review of gwcelery. Also, it will check the `NETRC` environment variable for you automatically.

Reply:
Yes, I pondered that then deliberately chose not to use it to eliminate a dependency. However, if there were significant security concerns about the builtin netrc (given that it now has the default file permission check) I can reconsider.